node/src/GlideClusterClient.ts

```diff
@@ -314,6 +318,151 @@ export class GlideClusterClient extends BaseClient {
+    ): command_request.ClusterScan {
+        const command = command_request.ClusterScan.create();
+        command.cursor = cursor;
+
+        if (options?.match) {
+            command.matchPattern = Buffer.from(options.match);
+        }
+
+        if (options?.count) {
+            command.count = options.count;
+        }
+
+        if (options?.type) {
+            command.objectType = options.type;
+        }
+
+        return command;
+    }
+
+    /**
+     * @internal
+     */
+    protected createClusterScanPromise(
+        cursor: ClusterScanCursor,
+        options?: ScanOptions & DecoderOption,
+    ): Promise<[ClusterScanCursor, GlideString[]]> {
+        // separate decoder option from scan options
+        const { decoder, ...scanOptions } = options || {};
+        const stringDecoder: boolean =
+            (decoder ?? this.defaultDecoder) === Decoder.String;
+
+        if (this.isClosed) {
+            throw new ClosingError(
+                "Unable to execute requests; the client is closed. Please create a new client.",
+            );
+        }
+
+        const cursorId = cursor.getCursor();
+        const command = this.scanOptionsToProto(cursorId, scanOptions);
+        return new Promise((resolve, reject) => {
+            const callbackIndex = this.getCallbackIndex();
+            this.promiseCallbackFunctions[callbackIndex] = [
+                (resolveAns) => {
+                    try {
+                        resolveAns = this.getValueToResolve(
+                            resolveAns,
+                            stringDecoder,
+                        );
+                        resolveAns[0] = new ClusterScanCursor(
+                            resolveAns[0].toString(),
+                        );
+                        resolve(resolveAns);
+                    } catch (err) {
+                        Logger.log(
+                            "error",
+                            "Decoder",
+                            `Decoding error: '${err}'`,
+                        );
+                        reject(err);
+                    }
+                },
+                reject,
+            ];
+            this.writeOrBufferCommandRequest(callbackIndex, command);
+        });
+    }
+
+    /**
+     * Incrementally iterates over the keys in the Cluster.
+     *
+     * This command is similar to the `SCAN` command but designed for Cluster environments.
+     * It uses a {@link ClusterScanCursor} object to manage iterations.
+     *
+     * For each iteration, use the new cursor object to continue the scan.
+     * Using the same cursor object for multiple iterations may result in unexpected behavior.
+     *
+     * For more information about the Cluster Scan implementation, see
+     * {@link https://github.com/valkey-io/valkey-glide/wiki/General-Concepts#cluster-scan | Cluster Scan}.
+     *
+     * This method can iterate over all keys in the database from the start of the scan until it ends.
```

Comment:
Add a note that this API collects data from all cluster slots

Reply:
I think it is important to mention, because this API does not have route argument.
It goes over all nodes (well, primary nodes - yes?), collects, and aggregates the data in runtime.

Reply:
If is refers to ReadFrom policy - this should be mentioned as well